### PR TITLE
feat: Inline plugins (plugins without running VM2)

### PR DESCRIPTION
**Pull request by oliverb123**
## Problem

Plugins run inside a VM, which incurs memory and cpu overhead, and code complexity. As many plugins are now considered trusted-source in cloud environments, we can inline the code for commonly used plugins instead, focusing for now on processEvent plugins, to get some improved performance in the ingestion pipeline.

### CODE REVIEW

**Review by tiina303 — APPROVED** [review]
Let's build out the frontend and backend changes on top of this to make sure we don't need anything else changed in the model to minimise number of migrations.

File: plugin-server/src/worker/vm/inline/noop.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    constructor(hub: Hub, pluginConfig: PluginConfig) {
+        this.hub = hub
+        this.config = pluginConfig
+        this.usedImports = new Set()
+
+        this.methods = {
+            processEvent: (event: PluginEvent) => {
+                return Promise.resolve(event)
+            },
+        }
```
[the remark this answers is not in the document]
Comment (oliverb123) [reply plugin-server/src/worker/vm/inline/noop.ts:33]: Gonna resolve elsewhere, I want to get this merged

File: plugin-server/src/types.ts [hunk 2]
```diff
@@ -392,9 +393,9 @@ export interface JobSpec {
 
 export interface Plugin {
     id: number
-    organization_id: string
+    organization_id: string | undefined
```
Comment (benjackwhite) [inline plugin-server/src/types.ts:396]: Nit: I generally prefer the optional marker
```suggestion
    organization_id?: string
```

File: plugin-server/src/worker/vm/inline/semver-flattener.ts
Comment (benjackwhite) [inline plugin-server/src/worker/vm/inline/semver-flattener.ts:1]: I was expecting to find a test for this tbh. At least one happy path test per inline plugin would be great

**Review by benjackwhite — APPROVED** [review]
This is super cool. Some really clean work here (especially when working with such a messy problem!) Only one suggestion regarding tests but not blocking.

We might want to be extra careful merging this. I haven't tried it locally so worth triple checking and if there is any doubts doing a canary style deploy (with support from infra)

File: plugin-server/src/worker/vm/inline/semver-flattener.ts
Comment (oliverb123) [reply plugin-server/src/worker/vm/inline/semver-flattener.ts:1]: I took the tests from the plugin repo and put them in `inline.test.ts` (the `semvar-flattener` one) - does this cover the kind of testing you mean, or something beyond that?

Comment (bretthoerner) [reply plugin-server/src/worker/vm/inline/semver-flattener.ts:1]: I'm not sure if it's what he meant, but seeing a functional test that actually calls an inline plugin would be nice (maybe I'm missing one?)

**Review by bretthoerner — APPROVED** [review]

**Pull request merged**